lib/cext/preprocess.rb

```diff
@@ -30,6 +35,36 @@
 
       "#{shim}(#{argc}, #{argv}, \"#{arity}\", "
     }
+  elsif line =~ /^(\s+)VALUE((\s+\w+\s*(\[\s*\d+\s*\]\s*)?,?)+);\s*$/
```

Comment:
The nested capture groups should be made non-capturing:
```ruby
 /^(\s+)VALUE((?:\s+\w+\s*(?:\[\s*\d+\s*\]\s*)?,?)+);\s*$/
```
You could use `/.../x` to make some extra space for readability if you want.